repos/system_upgrade/el9toel10/actors/scanpamuserdb/tests/test_scanpamuserdb.py

```diff
@@ -0,0 +1,23 @@
+import os
+
+import pytest
+
+from leapp.libraries.actor import scanpamuserdb
+
+CUR_DIR = os.path.dirname(os.path.abspath(__file__))
+
+
+def test_parse_file():
+    file = scanpamuserdb.parse_file(os.path.join(CUR_DIR, 'files/pam_userdb_missing'))
+    assert file is None
+    file = scanpamuserdb.parse_file(os.path.join(CUR_DIR, 'files/pam_userdb_basic'))
+    assert file == '/tmp/db1'
+    file = scanpamuserdb.parse_file(os.path.join(CUR_DIR, 'files/pam_userdb_complete'))
+    assert file == '/tmp/db2'
+
+
+def test_parse_folder():
```

Comment:
While it is theoretically possible not to use this directory, at least in the RHEL case it would not make sense because all applications that provide authentication services install their configuration files there. At the distribution level we force the files to be installed there and changing it could have catastrophic results for the user.

Also, I don't know of any customer who has changed this, so I would prefer not to test this case.

Reply:
In that case I agree